packages/airnode-node/test/e2e/requester-fulfill.feature.ts

```diff
@@ -65,24 +65,31 @@ it('should call fail function on AirnodeRrp contract and emit FailedRequest if r
 
 it('submits fulfillment with the gas price overridden', async () => {
   const requestedGasPrice = '99999999999'; // wei
-  const gasOverrideParameters = [
+  const overrideParameters = [
     { type: 'string32', name: 'from', value: 'ETH' },
     { type: 'string32', name: 'to', value: 'USD' },
     { type: 'string32', name: '_type', value: 'int256' },
     { type: 'string32', name: '_path', value: 'result' },
     { type: 'string32', name: '_times', value: '100000' },
     { type: 'string32', name: '_gasPrice', value: requestedGasPrice },
   ];
-  const requests = [operation.buildFullRequest({ parameters: gasOverrideParameters })];
+  const requests = [operation.buildFullRequest({ parameters: overrideParameters })];
   const { provider, deployment } = await deployAirnodeAndMakeRequests(__filename, requests);
 
   await startCoordinator();
 
   const logs = await fetchProviderLogs(provider, deployment.contracts.AirnodeRrp);
-  const fulfillmentLog = logs.find(async (log) => {
-    const tx = await provider.getTransaction(log.transactionHash);
-    tx.gasPrice?.toString() === requestedGasPrice;
-  });
 
-  expect(fulfillmentLog).not.toBeUndefined();
+  const filteredTxs = await Promise.all(
+    logs.map(async (log) => {
+      const tx = await provider.getTransaction(log.transactionHash);
+      if (tx.gasPrice && tx.gasPrice.toString() === requestedGasPrice) {
+        return tx;
+      }
+      return null;
+    })
+  ).then((logs) => logs.filter((log) => log));
```

Comment:
Even though the code is working this way I would prefer avoiding mixing `await` and `then`.

```suggestion
  const filteredTxs = (await Promise.all(
    logs.map(async (log) => {
      const tx = await provider.getTransaction(log.transactionHash);
      if (tx.gasPrice && tx.gasPrice.toString() === requestedGasPrice) {
        return tx;
      }
      return null;
    })
  )).filter((log) => log));
```

Reply:
Great- included in a fixup commit with you as co-author